lib/internal/url.js

```diff
@@ -840,6 +840,9 @@ class URL {
       if (path.length > 0) {
         try {
           const out = new URL(path);
+          if(out.protocol !== 'https:') {
+            return 'null';
+          }
```

Comment:
The standard says that 

> If pathURL’s scheme is not "http" and not "https", then return a new opaque origin.

For performance reasons, we avoid having string comparisons and calling getters which runs a string.prototype.slice. We can however use `scheme_type` numeric value. 

If you look into the declaration of `scheme_type` variable in `lib/internal/url.js`, you can find the following comment:

```
  /**
   * Refers to `ada::scheme::type`
   *
   * enum type : uint8_t {
   *   HTTP = 0,
   *   NOT_SPECIAL = 1,
   *   HTTPS = 2,
   *   WS = 3,
   *   FTP = 4,
   *   WSS = 5,
   *   FILE = 6
   * };
   * @type {number}
   */
```

For this particular case: we need to change line 846 to:

```
if (out.#context.scheme_type != 0 && != 2)
```

Reply:
Ah, thank you. I did notice `out.#context.scheme_type !== 1` but didn't pay attention to the comment.

Reply:
@anonrig commit looks better, now?

Reply:
can this enum be exported to js in some way? would be good to not just have magic numbers floating around

Reply:
At the very least, please add a comment indicating where the numbers come from